app/controllers/plans_controller.rb

```diff
@@ -0,0 +1,62 @@
+class PlansController < ApplicationController
+  before_action :authenticate_user!
+  before_action :find_plan_by_id, only: [:show, :edit, :update, :destroy]
+
+  
+  def index
+    @plans = Plan.all
+  end
+
+  def new
+    @plan = Plan.new
+  end
+
+  def edit
+  end
+
+  def create
+    #@plan = Plan.new(plan_params)
+    @company_categories_id = CompanyCategory.where(company_id: params[:company_id], category_id: params[:category_id])
+    @plan = Plan.new(plan_params.merge({company_categories_id: @company_categories_id.ids.uniq.first}))
```

Comment:
@BharatSoni-BTC it will insert company categories id in plan table and it is require during policy creation during policy creation